### PR TITLE
Avoid realizing SyntaxReference[] in source methods

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Addresses about 0.5% of allocations in a simple typing/lightbulb scenario:

image

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@333fred @RikkiGibson ptal.

**Review by 333fred — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
@jaredpar ptal

File: src/Compilers/CSharp/Portable/Symbols/Symbol.cs [hunk 1]
```diff
@@ -839,6 +838,10 @@ public virtual bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedWith
             return false;
         }
 
+        protected static bool IsDefinedInSourceTree(SyntaxReference syntaxRef, SyntaxTree tree, TextSpan? definedWithinSpan)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Symbols/Symbol.cs:841]: It feels a little unclear from the parameter names that we're testing whether `syntaxRef` refers to a location within `tree`. Would it be reasonable to make this an internal method or extension on `SyntaxReference`?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Symbol.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Symbols/Symbol.cs:841]: Sure.  added to the list of followups.

**Pull request merged**